test/plugins/compliance_visitor_data_request_plugin_test.rb

```diff
@@ -0,0 +1,159 @@
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": @visitor_two.properties["email"]})
+        assert_difference "ActionMailer::Base.deliveries.size", +2 do
+            perform_enqueued_jobs do
+                @data_request_plugin.run
+                Sidekiq::Worker.drain_all
+            end 
+        end
+    end
+
+    test "Sender and recipient email addresses and body message are correct" do
+        visitor_email = @visitor_one.properties["email"]
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": visitor_email})
+
+        expected_message = @data_request_plugin.metadata["MESSAGE"]
+
+        perform_enqueued_jobs do
+            @data_request_plugin.run
+            Sidekiq::Worker.drain_all
+        end
+
+        sent_email = ActionMailer::Base.deliveries.last
+        assert(sent_email.from.include?(Subdomain.current.name))
+        assert(sent_email.from.include?(ENV["APP_HOST"]))
+        assert_equal [visitor_email], sent_email.to
+        assert(sent_email.parts.first.body.raw_source.include?(expected_message))
+    end
+
+    test "Should not scan excluded api namespaces" do
+        @data_request_plugin.metadata["EXCLUDE_API_NAMESPACES"] << @namespace_with_email_one.slug
+        @data_request_plugin.save
+        visitor_email = @visitor_one.properties["email"]
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": visitor_email})
+        # Should not send an email even if a submission was made under a namespace since it's excluded
+        assert_no_difference "ActionMailer::Base.deliveries.size" do
+            perform_enqueued_jobs do
+                @data_request_plugin.run
+                Sidekiq::Worker.drain_all
+            end
+        end
+    end
+
+    test "Email should contain the right number of CSV attachments" do
+        visitor_email = @visitor_one.properties["email"]
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": visitor_email})
+        ApiResource.create(api_namespace_id: @namespace_with_email_two.id, properties: {"name": "Test Name", "email": visitor_email})
+
+        perform_enqueued_jobs do
+            @data_request_plugin.run
+            Sidekiq::Worker.drain_all
+        end
+
+        # Visitor one making submissions under two namespaces, so the email should contain two attachments
+        sent_email = ActionMailer::Base.deliveries.last
+        assert_equal 2, sent_email.attachments.length
+
+        sent_email.attachments.each do |attachment|
+            assert(attachment.content_type.start_with?("text/csv"))
+        end
+    end
+
+    test "CSV file should have the right format" do
```

Comment:
you are not testing format in this test case, so it should be renamed to "CSV file should be named according to the API Namespace and have file extension .csv"